### PR TITLE
Add instrumentation for search queries

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.clubhouse.io/simpledotorg/story/2020/add-instrumentation-for-patient-search

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/patient/filter/WeightedLevenshteinSearch.kt [hunk 1]
```diff
@@ -3,7 +3,6 @@ package org.simple.clinic.patient.filter
 import info.debatty.java.stringsimilarity.CharacterInsDelInterface
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/patient/filter/WeightedLevenshteinSearch.kt:3]: I had no idea, this class was a thing 😮 . Is this ranking for results?

**Review by msasikanth — DISMISSED** [review]
LGTM

File: app/src/main/java/org/simple/clinic/patient/filter/WeightedLevenshteinSearch.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/patient/filter/WeightedLevenshteinSearch.kt:3]: Yup

**Comment by msasikanth** [comment]
> /instrumentedTests

FYI this triggers an completely different Action, so it won't update the required check status because it never ran.

**Comment by vinaysshenoy** [comment]
> > /instrumentedTests
> 
> FYI this triggers an completely different Action, so it won't update the required check status because it never ran.

Can you explain this, please? I see the `qa_android_tests` job running. 🤔

**Comment by vinaysshenoy** [comment]
https://github.com/simpledotorg/simple-android/actions/runs/409737643

**Comment by msasikanth** [comment]
> > > /instrumentedTests
> > 
> > 
> > FYI this triggers an completely different Action, so it won't update the required check status because it never ran.
> 
> Can you explain this, please? I see the `qa_android_tests` job running. 🤔

If you look at the actions page there are 2 sets of CI workflows that are running. One is for this [PR](https://github.com/simpledotorg/simple-android/actions/runs/409728116) and [another](https://github.com/simpledotorg/simple-android/actions/runs/409737643) one is for the comment command. When we marked a action as required for a PR merge we expect it to be started when the PR is created. But because that test now triggeres when we leave a comment command it longer updates the status of the original action flow that is started when we openend a PR.

**Comment by vinaysshenoy** [comment]
> > > > /instrumentedTests
> > > 
> > > 
> > > FYI this triggers an completely different Action, so it won't update the required check status because it never ran.
> > 
> > 
> > Can you explain this, please? I see the `qa_android_tests` job running. thinking
> 
> If you look at the actions page there are 2 sets of CI workflows that are running. One is for this [PR](https://github.com/simpledotorg/simple-android/actions/runs/409728116) and [another](https://github.com/simpledotorg/simple-android/actions/runs/409737643) one is for the comment command. When we marked a action as required for a PR merge we expect it to be started when the PR is created. But because that test now triggeres when we leave a comment command it longer updates the status of the original action flow that is started when we openend a PR.

Hmm, then the current implementation is not yet ready to use. We make to make running instrumented tests mandatory to merge a PR, but not necessarily want to run it every time a PR is updated. Until we figure out how to make that happen, I think we should revert to the older flow for now.

**Review by msasikanth — APPROVED** [review]

**Pull request merged**